cmd/tempo/app/modules.go

```diff
@@ -90,6 +91,30 @@ func (t *App) initServer() (services.Service, error) {
 	return s, nil
 }
 
+func (t *App) initInternalServer() (services.Service, error) {
+	DisableSignalHandling(&t.cfg.InternalServer.Config)
+	serv, err := server.New(t.cfg.InternalServer.Config)
+	if err != nil {
+		return nil, err
+	}
+
+	servicesToWaitFor := func() []services.Service {
+		svs := []services.Service(nil)
+		for m, s := range t.serviceMap {
+			// Server should not wait for itself.
+			if m != InternalServer {
```

Comment:
as written will the internal server wait on the server and vice versa? do we want that behavior?